core/services/run_manager.go

```diff
@@ -21,20 +21,26 @@ import (
 )
 
 var (
-	numberRunsExecuted = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_started",
-		Help: "The total number of runs that have run",
-	})
-	numberRunsResumed = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_resumed",
-		Help: "The total number of run resumptions",
-	})
-	numberRunsCancelled = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "run_manager_runs_cancelled",
-		Help: "The total number of run cancellations",
-	})
+	promTotalRunUpdates = promauto.NewCounterVec(prometheus.CounterOpts{
+		Name: "run_manager_status_update_total",
+		Help: "The total number of status updates for Job Runs",
+	},
+		[]string{"job_spec_id", "from_status", "status"},
+	)
 )
 
+func promInstrumentJobRunUpdate(from models.RunStatus) func(*models.JobRun) {
+	return func(run *models.JobRun) {
+		if run == nil {
```

Comment:
Why are you passing in a nil run? I'd prefer not to have this kind of defensive programming, maybe instead pass in the JobSpecID, and status and require them to not be nil.

Reply:
Done :+1: